include/fastgltf/math.hpp

```diff
@@ -825,6 +825,65 @@ namespace fastgltf::math {
 			vec<T, 4>(l.x(), l.y(), l.z(), 1.f));
 	}
 
+	/** Inverts a 4x4 matrix, if possible. Returns true if success, false if failure.  A pointer to the input can also safely be used as the output */
+	FASTGLTF_EXPORT template <typename T>
+	bool invert(const mat<T, 4, 4>& m, mat<T, 4, 4>* output) noexcept {
+		T a00 = m[0][0];
+		T a01 = m[0][1];
+		T a02 = m[0][2];
+		T a03 = m[0][3];
+		T a10 = m[1][0];
+		T a11 = m[1][1];
+		T a12 = m[1][2];
+		T a13 = m[1][3];
+		T a20 = m[2][0];
+		T a21 = m[2][1];
+		T a22 = m[2][2];
+		T a23 = m[2][3];
+		T a30 = m[3][0];
+		T a31 = m[3][1];
+		T a32 = m[3][2];
+		T a33 = m[3][3];
+
+		T b00 = a00 * a11 - a01 * a10;
+		T b01 = a00 * a12 - a02 * a10;
+		T b02 = a00 * a13 - a03 * a10;
+		T b03 = a01 * a12 - a02 * a11;
+		T b04 = a01 * a13 - a03 * a11;
+		T b05 = a02 * a13 - a03 * a12;
+		T b06 = a20 * a31 - a21 * a30;
+		T b07 = a20 * a32 - a22 * a30;
+		T b08 = a20 * a33 - a23 * a30;
+		T b09 = a21 * a32 - a22 * a31;
+		T b10 = a21 * a33 - a23 * a31;
+		T b11 = a22 * a33 - a23 * a32;
+
+		// Calculate the determinant
+		T det = b00 * b11 - b01 * b10 + b02 * b09 + b03 * b08 - b04 * b07 + b05 * b06;
+		if (det == T(0)) {
+			// Can not invert this matrix
+			return false;
+		}
+		det = T(1) / det;
```

Comment:
While yes, mul ops are generally faster than div, I'd prefer if this operator would be applied to the entire matrix at the end, making the code a little simpler to read. I guess the `mat::operator/` should maybe make use of this minor optimization.
You would need to change the following function, and you could include it in this PR if you want, however you need to be careful since this trick only works for floating point numbers! So I guess you'd need to use SFINAE or concepts to differentiate between the two cases.
https://github.com/spnda/fastgltf/blob/159eb795d25c4466d14d209942959326007a0766/include/fastgltf/math.hpp#L747-L752

Reply:
It's fairly limited as an optimization, especially on modern processors. I just recently looked into assembly for a microprocessor I was using and I think a single div took about 4 times as long at around 20 cycles, but this is a fairly old arm microprocessor so it's not all to comparable to modern x86 or armv8 chips